test/ui-automation/test/specs/vault-interaction-flow.js

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+"use strict";
+
+import {wallet} from "../helpers";
+
+const vault = new Map();
+vault.set("Test Vault 1", {
+    "@context":["https://w3id.org/wallet/v1",
+        "https://trustbloc.github.io/context/wallet/collections-v1.jsonld"],
+    "id":"1",
+    "name":"Test Vault 1",
+    "type":"Vault"
+});
+vault.set("Test Vault 2", {
+    "@context":["https://w3id.org/wallet/v1",
+        "https://trustbloc.github.io/context/wallet/collections-v1.jsonld"],
+    "id":"2",
+    "name":"Test Vault 2",
+    "type":"Vault"
+});
+vault.set("Test Vault 3", {
+    "@context":["https://w3id.org/wallet/v1",
+        "https://trustbloc.github.io/context/wallet/collections-v1.jsonld"],
+    "id":"3",
+    "name":"Test Vault 3",
+    "type":"Vault"
+});
+vault.set("Test Vault 4", {
+    "@context":["https://w3id.org/wallet/v1",
+        "https://trustbloc.github.io/context/wallet/collections-v1.jsonld"],
+    "id":"4",
+    "name":"Test Vault 4",
+    "type":"Vault"
+});
+
+describe("TrustBloc Wallet - Add/Rename/Delete Vault flow", () => {
+    const ctx = {
+        email: `ui-aut-${new Date().getTime()}@test.com`,
+    };
+    const invalidVaultName ='Vault!@';
```

Comment:
Could we add checks for more invalid vault names? Duplicated names, string made of spaces only, string with multiple white spaces in a row (we should truncate those into 1), leading, trailing spaces

Reply:
Duplicate name is added.